runtime/ts/particle-execution-context.ts

```diff
@@ -116,16 +116,23 @@ export class ParticleExecutionContext {
          * renders content to the slot.
          */
         render(content) {
-          if (content.template && this.providedSlots.size > 0) {
+          // TODO: This logic should live in dom-particle and referencing slots by name should be deprecated for a special '{{slot:name}}' syntax.
+          if (this.providedSlots.size > 0) {
             content = {...content};
-            if (typeof content.template === 'string') {
-              content.template = this.substituteSlotNamesForIds(content.template);
-            } else {
-              content.template = Object.entries(content.template).reduce(
-                  (templateDictionary, [templateName, templateValue]) => {
-                    templateDictionary[templateName] = this.substituteSlotNamesForIds(templateValue);
-                    return templateDictionary;
-                  }, {});
+
+            content.model = {...content.model || {}};
+            this.providedSlots.forEach((slotId, slotName) => content.model[`__provided_slot.${slotName}`] = slotId);
```

Comment:
__provided_slot is cumbersome and difficult to read. Could we go with something more succinct?

Reply:
Maybe $slot, e.g. `$slot.action`? Dollar sign signifying a special field.

Reply:
I've moved to {{$localName}} in the end.